File: python/0_setup.py
import dcss_api

def main():
    # Connect to DCSS Webtile
    webtile = dcss_api.WebtilePy("ws://localhost:8080/socket", 100, "0.32")

    # Empty message queue
    while (message := webtile.get_message()) != None:
        pass

    # Register usernames for tests
    webtile.register_account("Username", "Password", None)
    webtile.register_account("Username2", "Password", None)
    webtile.register_account("Username3", "Password", None)
    webtile.register_account("Username4", "Password", None)

    # Disconnect from webtile
    webtile.disconnect()

if __name__ == "__main__":
    main()

File: dcss-api/examples/0_setup.rs
extern crate dcss_api;

use dcss_api::Webtile;

fn main() {
    // Connect to DCSS Webtile
    let mut webtile =
        Webtile::connect("ws://localhost:8080/socket", 100, "0.32").expect("Failed to connect");

    // Empty message queue;
    while webtile.get_message().is_some() {}

    // Register usernames for tests
    webtile
        .register_account("Username", "Password", None)
        .expect("Failed to register");

    // For "log in" test
    for i in 1..4 {
        webtile
            .register_account(format!("Username{i}").as_str(), "Password", None)
            .expect("Failed to register");
    }

    // For "tiles" test
    for i in 1..4 {
        webtile
            .register_account(format!("Tiles{i}").as_str(), "Password", None)
            .expect("Failed to register");
    }

    // For "monsters" test
    for i in 1..25 {
        webtile
            .register_account(format!("Monsters{i}").as_str(), "Password", None)
            .expect("Failed to register");
    }

    // Disconnect from webtile
    webtile.disconnect().expect("Failed to disconnect");
}
